src/blueprint/definitions.rs

```diff
@@ -344,8 +344,8 @@ pub fn update_cache_resolvers<'a>(
 ) -> TryFold<'a, (&'a ConfigModule, &'a Field, &'a config::Type, &'a str), FieldDefinition, String>
 {
     TryFold::<(&ConfigModule, &Field, &config::Type, &str), FieldDefinition, String>::new(
-        move |(_config, field, _, _name), mut b_field| {
-            if let Some(config::Cache { max_age }) = field.cache.as_ref() {
+        move |(_config, field, typ, _name), mut b_field| {
+            if let Some(config::Cache { max_age }) = field.cache.as_ref().or(typ.cache.as_ref()) {
```

Comment:
We should have a unit test for it. I think the execution spec can be made to fail if an IO is made more than.